### PR TITLE
PYIC-5212: add welesh translation for payslip NI page

### PR DESCRIPTION
**Pull request by shivanshuit914**
## Proposed changes

- Add welesh translation for payslip NI page
- Fix page title

### What changed

- Welsh translation yml files.
- Fixed page title in view using correct property.

### Why did it change

- To support welsh translation.

### Issue tracking

- [PYIC-5212](https://govukverify.atlassian.net/browse/PYIC-5212)
Screenshot 2024-03-05 at 11 12 33

[PYIC-5212]: https://govukverify.atlassian.net/browse/PYIC-5212?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by DanCorderIPV — DISMISSED** [review]

File: src/locales/cy/pages.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 intro:
   h1: Intro
 rti-payslip-national-insurance:
-  title: Enter the amount of National Insurance you paid from a recent payslip
-  inset: Use a payslip dated after {{ threeMonthsAgo }} to answer this question
+  title: Rhowch swm yr Yswiriant Gwladol a dalwyd gennych o slip cyflog diweddar
+  inset: Defnyddiwch slip cyflog wedi'i ddyddio ar ôl {{ threeMonthsAgo }} i ateb y cwestiwn hwn
```
Comment (blakeyp) [inline src/locales/cy/pages.yml:5]: We'll need to make sure the date (month) gets translated too - we should handle that in [PYIC-5211](https://govukverify.atlassian.net/browse/PYIC-5211)

File: src/locales/cy/pages.yml [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
   content:
-    - Tell us the amount of National Insurance you paid.
-    - You can usually find the amount next to any pensions or PAYE/income tax deductions on your payslip.
-    - Enter the amount in pounds and pence, exactly as it appears on your payslip.
-  abandon: "<a href='/kbv/abandon' class='govuk-link' data-id='abandon'>I need to prove my identity another way</a>"
+    - Dywedwch wrthym faint o Yswiriant Gwladol rydych wedi'i dalu.
+    - Fel arfer, gallwch ddod o hyd i'r swm wrth ymyl unrhyw bensiynau neu ddidyniadau TWE/Treth Incwm ar eich slip cyflog.
+    - Rhowch y swm mewn punnoedd a cheiniogau yn union fel y mae'n ymddangos ar eich slip cyflog.
+  abandon: "<a href='/kbv/abandon' class='govuk-link' data-id='abandon'>Mae angen i mi brofi fy hunaniaeth mewn ffordd arall</a>"
```
Comment (blakeyp) [inline src/locales/cy/pages.yml:10]: `abandon` will be shared across several pages, could we pull it out of this page and put it under a `shared` key or something

File: src/views/kbv/single-amount-question.njk [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 {% set gtmJourney = "kbv-hmrc - middle" %}
 
-{% set hmpoTitle = question.key %}
+{% set hmpoTitle = translate("pages.rti-payslip-national-insurance.title") %}
 
 {% block mainContent %}
   <h1 class="govuk-heading-l"> {{ translate("pages.rti-payslip-national-insurance.title") }} </h1>
```
Comment (blakeyp) [inline src/views/kbv/single-amount-question.njk:10]: As this template will be shared across multiple questions, we'll need a way of passing in the question key for the translations, maybe using `presenters` to build the content instead

File: src/app/kbv/controllers/single-amount-question.js [hunk 4]
```diff
@@ -15,15 +15,29 @@ class SingleAmountQuestionController extends BaseController {
         return callback(err, locals);
       }
 
+      if (req.session.question?.info?.months) {
+        locals.threeMonthsAgo = moment()
```
Comment (blakeyp) [inline src/app/kbv/controllers/single-amount-question.js:19]: Maybe `locals.dynamicDate` given the 3 months comes from the backend? (and the template doesn't care about this context)

File: src/app/kbv/controllers/single-amount-question.js [hunk 5]
```diff
@@ -15,15 +15,29 @@ class SingleAmountQuestionController extends BaseController {
...
+          req.session.question,
+          req.translate
+        ),
+        inset: presenters.questionToInset(
+          req.session.question,
+          req.translate,
+          locals
+        ),
+        title: presenters.questionToTitle(req.session.question, req.translate),
+        prefix: req.session.question?.info?.prefix || " ",
```
Comment (blakeyp) [inline src/app/kbv/controllers/single-amount-question.js:38]: Is this used?

File: src/views/kbv/single-amount-question.njk [hunk 6]
```diff
@@ -4,16 +4,16 @@
 
 {% set gtmJourney = "kbv-hmrc - middle" %}
 
-{% set hmpoTitle = question.key %}
+{% set hmpoTitle = translate("pages.rti-payslip-national-insurance.title") %}
```
Comment (blakeyp) [inline src/views/kbv/single-amount-question.njk:7]: should be `question.title` ?

File: src/app/kbv/controllers/single-amount-question.js [hunk 7]
```diff
@@ -15,15 +15,29 @@ class SingleAmountQuestionController extends BaseController {
         return callback(err, locals);
       }
 
+      if (req.session.question?.info?.months) {
+        locals.threeMonthsAgo = moment()
+          .subtract(req.session.question?.info?.months, "months")
+          .locale(req.lang)
+          .format("DD MMMM YYYY");
```
Comment (blakeyp) [inline src/app/kbv/controllers/single-amount-question.js:22]: We could abstract this part to a presenter, then we can test it in isolation (including test for welsh) and stub it in the controller tests?

File: src/locales/cy/pages.yml [hunk 8]
```diff
@@ -1,10 +1,11 @@
 intro:
   h1: Intro
+shared:
+  abandon: "<a href='/kbv/abandon' class='govuk-link' data-id='abandon'>Mae angen i mi brofi fy hunaniaeth mewn ffordd arall</a>"
 rti-payslip-national-insurance:
-  title: Enter the amount of National Insurance you paid from a recent payslip
-  inset: Use a payslip dated after {{ threeMonthsAgo }} to answer this question
+  title: Rhowch swm yr Yswiriant Gwladol a dalwyd gennych o slip cyflog diweddar
```
Comment (blakeyp) [inline src/locales/cy/pages.yml:6]: We may need to add a separate title and h1 here, just awaiting confirmation from UCD

Comment (blakeyp) [reply src/locales/cy/pages.yml:6]: Had confirmation we don't need to (title and h1 are the same)

**Review by blakeyp — APPROVED** [review]

**Review by merlinc — CHANGES_REQUESTED** [review]
Could we get better commit hygiene as per [GDS Way](https://gds-way.digital.cabinet-office.gov.uk/standards/source-code/working-with-git.html#commits)

Many of these commits are doing multiple things, and fixup commits (such as "fix linting") should be included into the other changes that you have done.

**Review by blakeyp — APPROVED** [review]

**Pull request merged**